src/libraries/navdrawer/navdrawer.js

```diff
@@ -3,358 +3,369 @@
-                        dom.removeEventListener(edgeContainer, 'touchend', onEdgeTouchEnd, {
+                        dom.removeEventListener(this.edgeContainer, 'touchend', this.onEdgeTouchEnd.bind(this), {
                             passive: true
                         });
-                        dom.removeEventListener(edgeContainer, 'touchcancel', onEdgeTouchEnd, {
+                        dom.removeEventListener(this.edgeContainer, 'touchcancel', this.onEdgeTouchEnd.bind(this), {
                             passive: true
                         });
                     }
                 }
             }
         }
-    };
+    }
 
-    TouchMenuLA.prototype.initialize = function () {
-        options = Object.assign(defaults, options || {});
+    initialize() {
+        const options = Object.assign(this.defaults, this.options || {});
 
         if (browser.edge) {
             options.disableEdgeSwipe = true;
         }
 
-        self.initElements();
+        this.initElements();
 
         if (browser.touch) {
-            dom.addEventListener(options.target, 'touchstart', onMenuTouchStart, {
+            dom.addEventListener(options.target, 'touchstart', this.onMenuTouchStart.bind(this), {
                 passive: true
             });
-            dom.addEventListener(options.target, 'touchmove', onMenuTouchMove, {
+            dom.addEventListener(options.target, 'touchmove', this.onMenuTouchMove.bind(this), {
                 passive: true
             });
-            dom.addEventListener(options.target, 'touchend', onMenuTouchEnd, {
+            dom.addEventListener(options.target, 'touchend', this.onMenuTouchEnd.bind(this), {
                 passive: true
             });
-            dom.addEventListener(options.target, 'touchcancel', onMenuTouchEnd, {
+            dom.addEventListener(options.target, 'touchcancel', this.onMenuTouchEnd.bind(this), {
                 passive: true
             });
-            dom.addEventListener(mask, 'touchstart', onBackgroundTouchStart, {
+            dom.addEventListener(this.mask, 'touchstart', this.onBackgroundTouchStart.bind(this), {
                 passive: true
             });
-            dom.addEventListener(mask, 'touchmove', onBackgroundTouchMove, {});
-            dom.addEventListener(mask, 'touchend', onBackgroundTouchEnd, {
+            dom.addEventListener(this.mask, 'touchmove', this.onBackgroundTouchMove.bind(this), {});
+            dom.addEventListener(this.mask, 'touchend', this.onBackgroundTouchEnd.bind(this), {
                 passive: true
             });
-            dom.addEventListener(mask, 'touchcancel', onBackgroundTouchEnd, {
+            dom.addEventListener(this.mask, 'touchcancel', this.onBackgroundTouchEnd.bind(this), {
                 passive: true
             });
         }
 
-        self.clickMaskClose();
-    };
-
-    return new TouchMenuLA();
+        this.mask.addEventListener('click', this.clickMaskClose.bind(this));
```

Comment:
I believe that this `bind` can be left as is, because it won't be removed at runtime. But, at the same time, can be replaced as others.